metaserver/src/main/scala/scala/meta/languageserver/providers/DocumentFormattingProvider.scala

```diff
@@ -2,25 +2,40 @@ package scala.meta.languageserver.providers
 
 import java.nio.file.Files
 import scala.meta.languageserver.Formatter
+import scala.meta.languageserver.Configuration
 import com.typesafe.scalalogging.LazyLogging
 import langserver.messages.DocumentFormattingResult
 import langserver.types.Position
 import langserver.types.Range
 import langserver.types.TextEdit
 import org.langmeta.inputs.Input
 import org.langmeta.io.AbsolutePath
+import monix.reactive.Observable
+import monix.eval.Task
+
+class DocumentFormattingProvider(configuration: Observable[Configuration], cwd: AbsolutePath) extends LazyLogging {
+
+  private lazy val scalafmt = Formatter.classloadScalafmt("1.3.0")
+
+  private def formatterFromConfiguration: Task[Formatter] = configuration.lastL.map { conf =>
+    if (conf.scalafmt.enable) scalafmt
+    else Formatter.noop
+  }
+
+  private def scalafmtConfigFromConfiguration: Task[AbsolutePath] = configuration.lastL.map { conf =>
+    cwd.resolve(conf.scalafmt.confPath)
+  }
 
-object DocumentFormattingProvider extends LazyLogging {
   def format(
-      input: Input.VirtualFile,
-      scalafmt: Formatter,
-      cwd: AbsolutePath
-  ): DocumentFormattingResult = {
+      input: Input.VirtualFile
+  ): Task[DocumentFormattingResult] = for {
+    formatter <- formatterFromConfiguration
+    config <- scalafmtConfigFromConfiguration
```

Comment:
`Task[A].zip(Task[B])`?

Reply:
I find forcomprehension more readable and scalable than tupling things up. Unless there's a strong penalty in flatMapping over tasks

Reply:
for comprehension is fine, I agree it's more readable, wish we had ApplicativeDo in Scala 😏 https://www.microsoft.com/en-us/research/publication/desugaring-haskells-do-notation-into-applicative-operations/

Reply:
I agree it more easily conveys the fact that I want to call A, then once A has returned call B.  But given that’s not what we want here, I think @olafurpg  version does convey the fact that these two things have no dependency on each other well.  So....sort of agree 🤷‍♂️

Reply:
Done, I've used zip.